### PR TITLE
Add `stream.relayTo()` and fix relay teardown

### PR DESCRIPTION
**Pull request by kasperisager**

### CODE REVIEW

**Comment by kasperisager** [comment]
@mafintosh This is ready for review.

File: src/udx.c [hunk 1]
```diff
@@ -1642,23 +1642,27 @@ udx_stream_destroy (udx_stream_t *handle) {
...
-  if (handle->relay_to == NULL) {
-    udx_packet_t *pkt = malloc(sizeof(udx_packet_t));
+  if (handle->relay_to != NULL) {
+    handle->status |= UDX_STREAM_DESTROYED;
+    close_maybe(handle, 0);
+    return 0;
+  }
 
-    uv_buf_t buf = uv_buf_init(NULL, 0);
+  udx_packet_t *pkt = malloc(sizeof(udx_packet_t));
```
Comment (mafintosh) [inline src/udx.c:1651]: who frees this?

Comment (kasperisager) [reply src/udx.c:1651]: The normal packet sending flow; the line was merely moved out of a conditional that is now handled above.

Comment (mafintosh) [reply src/udx.c:1651]: ahhh, freaking github

**Pull request merged**